lib/go/thrift/simple_server.go

```diff
@@ -149,8 +151,10 @@ func (p *TSimpleServer) Serve() error {
 }
 
 func (p *TSimpleServer) Stop() error {
-	p.quit <- struct{}{}
-	p.serverTransport.Interrupt()
+	if atomic.CompareAndSwapInt64(&p.stopped, 0, 1) {
```

Comment:
When I looked at the go documentation for CompareAndSwapInt64 the documentation states that it returns the old value, so I assume you want the quit and interrupt code to run when this method returns 0, meaning it is the first time it has run.